test/bigtable/index.js

```diff
@@ -0,0 +1,313 @@
+    });
+  });
+
+  describe('formatTableName_', function() {
+    it('should return the last section of a formatted table name', function() {
+      var fakeTableName = 'projects/p/zones/z/clusters/c/tables/my-table';
+      var formatted = Bigtable.formatTableName_(fakeTableName);
+
+      assert(formatted, 'my-table');
+    });
+
+    it('should do nothing if the table is name is not formatted', function() {
+      var fakeTableName = 'my-table';
+      var formatted = Bigtable.formatTableName_(fakeTableName);
+
+      assert(formatted, fakeTableName);
+    });
+  });
+
+  describe('createTable', function() {
+    var TABLE_ID = 'my-table';
+
+    it('should provide the proper request options', function(done) {
+      bigtable.request = function(protoOpts, reqOpts) {
+        assert.deepEqual(protoOpts, {
+          service: 'BigtableTableService',
+          method: 'createTable'
+        });
+
+        assert.strictEqual(reqOpts.name, CLUSTER_NAME);
+        assert.strictEqual(reqOpts.tableId, TABLE_ID);
+        done();
+      };
+
+      bigtable.createTable(TABLE_ID, assert.ifError);
+    });
+
+    it('should not modify the original options', function(done) {
+      var options = {
+        a: 'a',
+        b: 'b'
+      };
+
+      var originalOptions = extend({}, options);
+
+      bigtable.request = function(protoOpts, reqOpts) {
+        assert.notStrictEqual(reqOpts, options);
+        assert.deepEqual(options, originalOptions);
+        done();
+      };
+
+      bigtable.createTable(TABLE_ID, options, assert.ifError);
+    });
+
+    it('should set the current operation', function(done) {
+      var options = {
+        operation: 'abc'
+      };
+
+      bigtable.request = function(protoOpts, reqOpts) {
+        assert.strictEqual(reqOpts.table.currentOperation, options.operation);
+        done();
+      };
+
+      bigtable.createTable(TABLE_ID, options, assert.ifError);
+    });
+
+    it('should set the initial split keys', function(done) {
+      var options = {
+        splits: ['a', 'b']
+      };
+
+      bigtable.request = function(protoOpts, reqOpts) {
+        assert.strictEqual(reqOpts.initialSplitKeys, options.splits);
+        done();
+      };
+
+      bigtable.createTable(TABLE_ID, options, assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+
+      bigtable.request = function(protoOpts, reqOpts, callback) {
+        callback(err);
+      };
+
+      bigtable.createTable(TABLE_ID, function(err_) {
+        assert.strictEqual(err, err_);
+        done();
+      });
+    });
+
+    it('should return a Table object', function(done) {
+      var response = {
+        name: TABLE_ID
+      };
+
+      bigtable.request = function(p, r, callback) {
+        callback(null, response);
+      };
+
+      bigtable.createTable(TABLE_ID, function(err, table, apiResponse) {
+        assert.ifError(err);
+        assert(table instanceof FakeTable);
```

Comment:
Stub `bigtable.table` instead and see that it was called correctly. The return value from it should be compared with this `table` argument.